middleware/decrypt.py

```diff
@@ -0,0 +1,76 @@
+"""Identify and decrypt Crypt4GH keys and files.
+
+Decrypts crypt4GH files given a list of files and places the output in a specified
+directory. If a file is not Crypt4GH-encrypted, it moves the file to the output directory
+without alteration. If a Crypt4GH file is included, the private key associated with that
+file must be provided.
+
+Example:
+    python3 decrypt.py --output-dir /outputs/ file.txt file.c4gh sk.sec pk.pub
+"""
+import shutil
+from argparse import ArgumentParser
+from pathlib import Path
+from crypt4gh.lib import decrypt  # type: ignore
+from crypt4gh.keys import get_private_key  # type: ignore
+
+
+def get_private_keys(file_paths: list[Path]) -> list[bytes]:
+    """Retrieve private keys from a list of files.
+
+    Args:
+        file_paths (list[Path]): A list of file paths.
+
+    Returns:
+        list[bytes]: A list of retrieved private keys as byte objects.
+    """
+    private_keys = []
+    for file_path in file_paths:
+        try:
+            key = get_private_key(Path(file_path), lambda x: '')
+            private_keys.append(key)
+        except ValueError:
+            continue
+    return private_keys
+
+
+def decrypt_files(
+        file_paths: list[Path],
+        private_keys: list[bytes]):
+    """Decrypt files and save to specified output directory.
+
+    Args:
+        file_paths (list[Path]): A list of file paths.
+        private_keys (list[bytes]): A list of private keys as byte objects.
+
+    Raises:
+        ValueError: If no private key for a Crypt4GH file is provided.
+    """
+    for file_path in file_paths:
+        with open(file_path, "rb") as f_in:
+            try:
+                with open('temp_file', "wb") as f_out:
+                    decrypt([(0, pk, "") for pk in private_keys], f_in, f_out)
```

Comment:
What is the reason for trying to decrypt all files with all keys? In any case, a comment explaing the first argument to `decrypt()` might be useful.

Also, for better readibility/clarity, I think it's useful to explicitly list arg names in function/method calls, especially if there's more than one.

Reply:
In the crypt4gh library there isn't a way that I could find that would allow me to find the associated key with each crypt4gh file. The header is decrypted through iterating through a list of keys [here](https://github.com/EGA-archive/crypt4gh/blob/7dc0d8b288397f006d7472c572bf4c82bfa2a3a4/crypt4gh/header.py). Theoretically I could  require the user to specify which secret key is associated with each file through environment variables, but I'm not sure the reduction in runtime is worth the extra configuration.

Reply:
Thanks, but I'm still confused :see_no_evil: If anyone can just find the "secret" keys and decrypt the files, where's the value of encryption? Or is this just a placeholder of sorts, for testing?

Reply:
Ahhh yeah that was what prompted [this](https://elixircloud.slack.com/archives/C04RLFJNF7U/p1716572664868359) discussion about getting the keys in the executor. My plan is to have a separate executor that retrieves the keys (which I haven't figured out yet), so the user doesn't actually pass in the keys in the request. Either way, what the decryption executor is going to handle is a file path to the secret key (whether that be the user's key obtained through some sort of vault or an ephemeral key generated by the previous executor).

Reply:
Got it, thanks :)